pkg/controller/job/job_controller.go

```diff
@@ -772,6 +766,8 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 
 	// if job was finished previously, we don't want to redo the termination
 	if util.IsJobFinished(&job) {
+		// The job shouldn't be marked as finished until all pod finalizers are removed.
```

Comment:
This is generally true and protected by the code here: https://github.com/kubernetes/kubernetes/blob/abc056843c030ead11551e13171a4447701db24e/pkg/controller/job/job_controller.go#L1398-L1403.

So, we may not need this code here, unless we motivate it by being a safe-guard in case of a bug, or user setting terminal condition on the Job via API.

Reply:
It is a safeguard